plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/PipelineArgumentsTab.java

```diff
@@ -399,36 +393,23 @@ public String getName() {
   }
 
   private void updatePipelineOptionsForm() {
-    final SettableFuture<Map<PipelineOptionsType, Set<PipelineOptionsProperty>>> optionsHierarchyFuture =
-        SettableFuture.create();
-    optionsHierarchyFuture.addListener(new Runnable() {
-      @Override
-      public void run() {
-        if (internalComposite.isDisposed()) {
-          return;
-        }
-        BusyIndicator.showWhile(internalComposite.getDisplay(), new Runnable() {
-          @Override
-          public void run() {
-            try {
-              pipelineOptionsForm.updateForm(launchConfiguration, optionsHierarchyFuture.get());
-              updateLaunchConfigurationDialog();
-            } catch (InterruptedException | ExecutionException ex) {
-              DataflowUiPlugin.logError(ex, "Exception while updating available Pipeline Options"); //$NON-NLS-1$
-            }
-          }
-        });
-      }
-    }, displayExecutor);
     try {
+      // TODO: use "Holder" instead of "AtomicReference" when Java 8 is available.
```

Comment:
I don't see a Holder class in Java 8?

Reply:
You're right. Can't find a good class. I think I should just update comment to say "this is just a holder; no atomicity is required."